LuaSkin/LuaSkin/Skin.m

```diff
@@ -294,6 +338,15 @@ - (void)checkArgs:(int)firstArg, ... {
                 break;
             case LUA_TNUMBER:
                 lsType = LS_TNUMBER;
+
+                if ((spec & lsType) && (spec & LS_TINTEGER)) {
+                    int isInteger ;
+                    lua_tointegerx(_L, idx, &isInteger) ;
+                    if (!isInteger) {
```

Comment:
Well, since it's set in the labels passed to `checkArgs:`, that's kind of the point... checkArgs throws lua errors when you pass in the wrong arguments.

However, I can see the argument that since we haven't done this in the past and because Lua does have a default behavior (it returns 0 for `lua_tointeger`) that doesn't throw an error (unless you explicitly use `luaL_checkinteger`), perhaps using `logError:` might be better.  It would be different than other argument errors, but perhaps kinder to pre-existing code. OTOH, if the module truly does want a proper integer, it will still have to use it's own integer check to know if 0 is because 0 was specified or because the number wasn't a proper integer.

I'll switch if you think it's a better approach.  Let me know.